packages/core/src/node/uploadContribution.ts

```diff
@@ -0,0 +1,62 @@
+/*
+* Copyright (C) 2017 Ericsson and others.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*/
+import { injectable, inject } from 'inversify';
+import { ILogger } from "../common";
+import { BackendApplicationContribution } from '../node';
+import * as express from 'express';
+import { Request } from 'express';
+import * as multer from 'multer';
+import * as fs from 'fs-extra';
+
+const storage = multer.diskStorage({
```

Comment:
Could you elaborate that it does?

Reply:
This backend module is using multer package to define where we want to upload the new files/folder. Using the "storage" from the package, it allows us to define the selected destination  (Not only a pre-defined location). When the end-user is dragging a file over a folder in the theia navigator, we record the destination and pass it to the backend. This location  is being handled by the multer.storage.destination. The second part of the multer.storage allows us to restore the original file name into the new destination.
There is other package allowing us to handle the upload file (ex:  body-parser and formidable),  ( https://www.npmjs.com/package/body-parser ), but it seems to me the one doing a good job and being simpler to use was multer

Reply:
I asked already, but maybe you missed it. Couldn't we just rely on the `FileSystem` instead?

Reply:
Using the FileSystem, so this file is removed from the PR.